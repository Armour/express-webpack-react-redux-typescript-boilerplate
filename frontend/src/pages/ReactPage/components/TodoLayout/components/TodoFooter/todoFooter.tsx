import React from 'react';
import { NamespacesConsumer } from 'react-i18next';

import TodoFilter from './components/TodoFilter';

export class TodoFooter extends React.Component {
  public render() {
    return (
      <NamespacesConsumer ns='reactPage'>
        {(t) => (
          <div className='todo-footer'>
            <TodoFilter filter='SHOW_ALL'>
              {t('todoLayout.todoFooter.all')}
            </TodoFilter>
            <TodoFilter filter='SHOW_ACTIVE'>
              {t('todoLayout.todoFooter.active')}
            </TodoFilter>
            <TodoFilter filter='SHOW_COMPLETED'>
              {t('todoLayout.todoFooter.completed')}
            </TodoFilter>
          </div>
        )}
      </NamespacesConsumer>
    );
  }
}

export default TodoFooter;
