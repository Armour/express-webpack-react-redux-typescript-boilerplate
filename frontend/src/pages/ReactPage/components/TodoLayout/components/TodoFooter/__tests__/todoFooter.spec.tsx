import React from 'react';
import TestRenderer from 'react-test-renderer';

import { TodoFooter } from '../todoFooter';

jest.mock('../components/TodoFilter', () => 'TodoFilter');

describe('TodoFooter', () => {
  it('should renders correctly', () => {
    const renderer = TestRenderer.create(
      <TodoFooter />,
    );
    expect(renderer).toMatchSnapshot();
    renderer.unmount();
  });
});
