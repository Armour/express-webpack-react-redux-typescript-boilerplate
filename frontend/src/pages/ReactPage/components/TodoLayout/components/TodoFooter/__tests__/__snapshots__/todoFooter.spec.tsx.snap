// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFooter should renders correctly 1`] = `
<div
  className="todo-footer"
>
  <TodoFilter
    filter="SHOW_ALL"
  >
    todoLayout.todoFooter.all
  </TodoFilter>
  <TodoFilter
    filter="SHOW_ACTIVE"
  >
    todoLayout.todoFooter.active
  </TodoFilter>
  <TodoFilter
    filter="SHOW_COMPLETED"
  >
    todoLayout.todoFooter.completed
  </TodoFilter>
</div>
`;
